sdk/core/core-client-rest/src/getClient.ts

```diff
@@ -59,9 +60,13 @@ export function getClient(
   }
 
   const { allowInsecureConnection, httpClient } = clientOptions;
+  const endpointUrl = endpoint ?? clientOptions.endpoint ?? clientOptions.baseUrl;
+  if (clientOptions.baseUrl && endpointUrl === clientOptions.baseUrl) {
+    logger.warning("The baseUrl parameter is deprecated. Please use endpoint instead.");
```

Comment:
Added the logger warning here, since we haven't added `@azure/logger` as a dependency of before, I wonder if you have any concerns here. @joheredi @xirzec  ?

Reply:
It looks like logger adds around 2kb to the bundle size. I would rather not add that just for this log. Can we do console.warn?

Reply:
I am not sure if console.warn will cause problems , if we are concerned about the bundle size, I would probably prefer to not take options.baseUrl into account, as it's not working in the core-client-rest today anyway ?

Reply:
Yeah that's fair